packages/workato-adapter/src/filters/cross_service/zendesk/reference_finder.ts

```diff
@@ -0,0 +1,234 @@
+              optionsByValue[input[field]],
+              path.createNestedID('input', field),
+            )
+          }
+        }
+      })
+    }
+
+    if (input.macro_ids !== undefined && input.macro_ids.id !== undefined) {
+      addPotentialIdReference(
+        indexedElements.elementsByInternalID.macros[input.macro_ids.id],
+        path.createNestedID('input', 'macro_ids', 'id')
+      )
+    }
+
+    if (input.group_id !== undefined) {
+      addPotentialIdReference(
+        indexedElements.elementsByInternalID.groups[input.group_id],
+        path.createNestedID('input', 'group_id')
+      )
+    }
+
+    if (input.brand_id !== undefined) {
+      addPotentialIdReference(
+        indexedElements.elementsByInternalID.brands[input.brand_id],
+        path.createNestedID('input', 'brand_id')
+      )
+    }
+
+    if (input.ticket_form_id !== undefined) {
+      addPotentialIdReference(
+        indexedElements.elementsByInternalID.ticketForms[input.ticket_form_id],
+        path.createNestedID('input', 'ticket_form_id')
+      )
+    }
+
+    if (BLOCK_NAMES_BY_TYPE.ticket.includes(name)) {
+      addCustomFieldsReferences({
+        indexKey: indexedElements.elementsByInternalID.ticketFields,
+        indexValue: indexedElements.ticketCustomOptionByFieldIdAndValue,
+        regex: ID_FIELD_REGEX,
+        fieldKeyConverter: match => Number(match),
+      })
+    } else if (BLOCK_NAMES_BY_TYPE.user.includes(name)) {
+      addCustomFieldsReferences({
+        indexKey: indexedElements.customFieldsByKey.user,
+        indexValue: indexedElements.customOptionsByFieldKeyAndValue.user,
+      })
+    } else if (BLOCK_NAMES_BY_TYPE.organization.includes(name)) {
+      addCustomFieldsReferences({
+        indexKey: indexedElements.customFieldsByKey.organization,
+        indexValue: indexedElements.customOptionsByFieldKeyAndValue.organization,
+      })
+    }
+
+    return references
+  }
+
+  const formulaFieldMatcher = createFormulaFieldMatcher(appName)
+
+  const formulaReferenceFinder: FormulaReferenceFinder = (value, path) => {
+    const potentialMatchGroups = formulaFieldMatcher(value)
+    return potentialMatchGroups.map(({ custom, block, field: fieldName }) => {
+      if (custom !== undefined && block !== undefined && zendeskBlocks.includes(block)) {
+        if (custom === 'custom') { // === 'ticket'. there is no ticket_fields only custom_fields
+          const fieldId = Number(fieldName)
+          return !_.isNaN(fieldId)
+            ? getFieldReference(fieldId, indexedElements.elementsByInternalID.ticketFields, path)
+            : undefined
+        }
+
+        if (custom === 'user') {
+          return getFieldReference(fieldName, indexedElements.customFieldsByKey.user, path)
+        }
+
+        if (custom === 'organization') {
+          return getFieldReference(fieldName, indexedElements.customFieldsByKey.organization, path)
+        }
+        log.debug(`Unknown field_${custom} in Zendesk Formula: ${value}`)
```

Comment:
please use 
```suggestion
        log.debug('Unknown field_%s in Zendesk Formula: %s', custom, value)
```
so that the string won't be calculated if the log leve is below debug